### DISCUSSION TITLE
How do you handle Fedora container images?

### DISCUSSION THREAD

**Discussion by SISheogorath** [renovatebot/renovate#14425] [Request Help] [answered]
Currently when using a Fedora base image for containers, e.g. `docker.io/library/fedora:35` renovare will immediately recommend zo upgrade to Fedora 36. But Fedora 36 is testing/beta and not considered stable.

While I could force renovate in it's config to only update to 35, I suddenly have to manage these updates myself again. So I wonder how you people manage Fedora image upgrades in an automated fashion?

**Comment by viceice** [comment]
Currently only possible via manual config updates or dependency dashboard approval.

We have a custom versioning for Ubuntu, where we only consider lts releases as stable updates. Maybe someone can crate something similar for fedora, but not sure how widely fedora is used on containers.

**Reply by JaneSoo** [reply]
@viceice Could you share how you did that? I tried to find config option, I only see `ignoreUnstable` which is by default already true.

**Reply by JaneSoo** [reply]
I found PR for ubuntu https://github.com/renovatebot/renovate/pull/7126/files maybe I could create similar MR for fedora?

**Reply by SISheogorath** [reply]
I think nowadays you can use https://docs.renovatebot.com/modules/datasource/endoflife-date/

**Reply by JaneSoo** [reply]
Thanks a lot @SISheogorath 
It seems to me that we need to add this `# renovate: datasource=endoflife-date depName=amazon-eks versioning=loose` to all the files we need to update. I have several docker files that are fedora-based and intend to have more in the future. Also, I was thinking of applying similar things to other distributions like centos as well. Thought, there is something that can be applied in general

**Comment by Blaimi** [comment, marked as answer]
I'm using this config:
```json
{
  "$schema": "https://docs.renovatebot.com/renovate-schema.json",
  "extends": [
    "config:recommended"
  ],
  "customManagers": [
    {
      "customType": "regex",
      "description": "Update Fedora base image based on endoflife-date",
      "fileMatch": [
        "(^|/|\\.)([Dd]ocker|[Cc]ontainer)file$",
        "(^|/)([Dd]ocker|[Cc]ontainer)file[^/]*$"
      ],
      "matchStrings": [
        "FROM quay.io/fedora/fedora:(?<currentValue>[^\\s]+).*\\n",
        "COPY --from=quay.io/fedora/fedora:(?<currentValue>[^\\s]+).*\\n"
      ],
      "versioningTemplate": "loose",
      "datasourceTemplate": "endoflife-date",
      "depNameTemplate": "fedora"
    }
  ],
  "packageRules": [
    {
      "matchDatasources": ["docker"],
      "matchDepNames": ["quay.io/fedora/fedora"],
      "enabled": false
    }
  ]
}
```

please note, that this kind of config does not support all the features of the `docker` datasource, e.g. digests and that the `docker` datasource has more advanced (and multiple) regexes to find versions.
